entrypoint.py

```diff
@@ -24,23 +26,128 @@ def age(self, birthdate):
+        curr_max_allowed = int(curr_max_allowed)
+        answers_list = self.create_list_of_answer(series)
+        for answers in enumerate(answers_list):
+            if len(answers) > curr_max_allowed:
+                return True
+        return False
+
+    def split_and_clean(self, curr_list_answers, token=r'[,=]'):
+        curr_list_answers_list = list(
+            curr_list_answers.str.split(token).values)
+        curr_list_answers_list = list(
+            map(str.strip, [item for sublist in curr_list_answers_list for item in sublist]))
+        return curr_list_answers_list
+
+    def check_categorical(self, series, measure_level):
+        curr_list_answers_list = self.split_and_clean(measure_level, r'[,=]')
+        for combined_value in series.values:
+            # Split the combined value based on ';'
+            values = combined_value.split(';')
+            for value in values:
+                if value not in curr_list_answers_list:
+                    return True
+        return False
+
+    def check_ordinal(self, series, measure_level):
+        curr_list_answers_list = self.split_and_clean(measure_level, r'[,]')
+        for value in series.values:
+            if value not in curr_list_answers_list:
+                return True
+        return False
 
-        for column in columns:
-            if column not in metadata_variables:
+    def check_numeric(self, series):
+        for value in series.values:
+            try:
+                float(value)  # Prova a convertire il valore in un numero float
+            except ValueError:
+                # Se si verifica un'eccezione, significa che il valore non può essere convertito in un numero
                 return True
+        return False
 
+    def check_boolean(self, series):
+        accepted_boolean_values = ["Yes", "Y", "No", "N"]
+        for value in series.values:
+            if value not in accepted_boolean_values:
+                return True
         return False
 
+    def check_text(self, series):
+        for value in series.values:
+            if not isinstance(value, str):
+                return True
+        return False
+
+    def check_data_type(self, series, json_response_df):
+        curr_data_type = json_response_df.loc[json_response_df['name'] == series.name, 'data_type'].values[0]
+        curr_list_answers = json_response_df.loc[json_response_df['name'] == series.name, 'measure_level']
+        curr_data_type = curr_data_type.lower()
+        if curr_data_type == "categorical":
+            return self.check_categorical(series, curr_list_answers)
+        elif curr_data_type == "ordinal":
+            return self.check_ordinal(series, curr_list_answers)
+        elif curr_data_type == "numeric":
+            return self.check_numeric(series)
+        elif curr_data_type == "boolean":
+            return self.check_boolean(series)
+        elif curr_data_type == "text":
+            return self.check_text(series)
+        else:
```

Comment:
This function covers `categorical, ordinal, numeric, boolean, text` data types. While creating variables, there are more data types which can be chosen `(decimal, string, date, etc).` Please, include them also in this check

Reply:
There are some data type that are redoundant, like string and text.
Decimal and number can be implemented as number for integer numbers and decimal for the float one.
If you think that this difference is important let me know.

Reply:
Yes, I agree there is no difference in the check, but if someone chooses string, there is no branch for that, which leads to else branch indicating that data is incorrect. So, if all of those datatypes remain as an option which can be chosen in UI, also the redundant one needs to be considered. I think just adding in the already existing corresponding branch is the easiest